src/Draco.Editor.Web/app/ts/cache.ts

```diff
@@ -15,8 +15,28 @@ export function getDownloadViewElement() {
 }
 
 export async function downloadAssemblies(cfg: unknown) {
+    try {
+        const cache = await caches.open('assembly-cache');
+        const result = await cache.match('appBuildDate');
+        const cachedDate = await result.text();
+        console.log('Current build: '+ buildDate+' cache build: '+cachedDate);
+
+        if (result == null || cachedDate != buildDate) {
+            console.log('Cache nuked.');
+            await caches.delete('assembly-cache');
+        }
+    } catch (e) {
+        console.log('Could not open cache: ', e);
+    }
+    try {
+        const cache = await caches.open('assembly-cache');
+        await cache.put('appBuildDate', new Response(buildDate));
```

Comment:
Can't this be in the previous try catch?

Reply:
Do i understand correctly that this fails if you nuked the cache before? if yes, where do you put the new date after you loaded the new compiler version into the cache?

Reply:
In certain scenario, calling caches.open just doesn't work some browser, for example when you are in private mode.